LongTasks/explainer.md

```diff
@@ -18,237 +18,215 @@ The Long Tasks API allows web developers to monitor long-running tasks on the ma
-|TaskAttributionTiming.containerSrc|Returns the container's src attribute.|Returns the worker script's URL|
-|TaskAttributionTiming.containerId|Returns the container's id attribute.|Always returns an empty string|
-|TaskAttributionTiming.containerName|Returns the container's name attribute.|Always returns an empty string|
+    // End measuring time
+    performance.mark(endMark);
+    performance.measure(measureName, startMark, endMark);
+
+    // Get the measurement
+    const duration = performance.getEntriesByName(measureName)[0].duration;
 
-For Web Workers, the containerType should return "worker" to clearly indicate the context. Additionally, containerSrc can return the URL of the worker script, helping developers differentiate between multiple workers.
+    // Send the result and duration back to the main thread
+    postMessage({no: e.data.no, res: fibo_num , duration: duration});
 
-## Monitoring multiple web workers
+    // Clear the marks and measurements to avoid memory leaks
+    performance.clearMarks(startMark);
+    performance.clearMarks(endMark);
+    performance.clearMeasures(measureName);
+};
+```
 
-The main thread can create multiple workers. Instead of setting up a `PerformanceObserver` for each worker, the main thread can gather performance entries from all workers and its own long tasks. Therefore, it is only necessary to set up the `PerformanceObserver` for the longtask type on the main thread to collect these entries:
+The results can vary due to randomness, but an example output might look like the following. In this case, task 4 is delayed because task 3 took 109.60ms to complete. Consequently, task 4 experienced a total duration of 73.90ms, including time spent waiting(48.40ms) in the message queue.
+
+```bash
+* Sending task 0 to the Fibonacci worker with input 32
+  * Result: Task 0 returned 2,178,309 in 29.40 ms. (message queue wait time:1.80 ms, task duration: 27.60 ms)
+* Sending task 1 to the Fibonacci worker with input 30
+  * Result: Task 1 returned 832,040 in 11.30 ms. (message queue wait time:0.70 ms, task duration: 10.60 ms)
+* Sending task 2 to the Fibonacci worker with input 31
+  * Result: Task 2 returned 1,346,269 in 17.60 ms. (message queue wait time:1.30 ms, task duration: 16.30 ms)
+* Sending task 3 to the Fibonacci worker with input 35
+  * Sending task 4 to the Fibonacci worker with input 32
+* Result: Task 3 returned 9,227,465 in 111.10 ms. (message queue wait time:1.50 ms, task duration: 109.60 ms)
+  * Result: Task 4 returned 2,178,309 in 73.90 ms. (message queue wait time:48.40 ms, task duration: 25.50 ms)
+```
+However, identifying whether task 3 qualifies as a "long task" is challenging without relying on the mark and measure APIs provided by the Performance interface.
 
-## Main script (main.js)
+In practice, we have observed long tasks lasting several minutes due to unexpected network delays or IndexedDB overhead.
```

Comment:
As also mentioned by Noam on email, I believe async delays in worker tasks such as due to network delays or slow IndexedDB responses are not in scope of what constitutes as a "long task", and that we're interested to know when the worker is actually performing synchronous compute on the thread because of which there are delays in processing other async events such as from postMessage.

Reply:
I agree with your suggestion. I will remove the sentence.